### PR TITLE
Nisha | Stage | Provide support of Entries export in Giddh

### PR DESCRIPTION
**Pull request by nisha-walkover**
https://app.clickup.com/t/86cvqx9c3

### CODE REVIEW

File: apps/web-giddh/src/app/daybook/daybook.component.ts [hunk 1]
```diff
@@ -379,55 +381,77 @@ export class DaybookComponent implements OnInit, OnDestroy {
             this.daybookQueryRequest.type = response.type;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/daybook/daybook.component.ts]: if (response && response !== 'close') {
            this.daybookQueryRequest.type = response.type;
            
Suggestion - check response so below all code work fine

File: apps/web-giddh/src/app/daybook/daybook.component.ts [hunk 2]
```diff
@@ -379,55 +381,77 @@ export class DaybookComponent implements OnInit, OnDestroy {
...
-            } else if (this.daybookExportRequestType === 'get') {
-                this.daybookService.ExportDaybook(null, this.daybookQueryRequest).pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            } else {
+                // for expanded option download
+                let exportBodyRequestObj: ExportBodyRequest = new ExportBodyRequest();
+                exportBodyRequestObj.from = this.daybookQueryRequest.from;
+                exportBodyRequestObj.to = this.daybookQueryRequest.to;
+                exportBodyRequestObj.fileType = this.fileTypeExpand;
+                exportBodyRequestObj.exportType = "ENTRIES_EXPORT";
+                let branchUniqueName = this.generalService.currentBranchUniqueName ? this.generalService.currentBranchUniqueName : (this.currentBranch) ? this.currentBranch?.uniqueName : "";
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/daybook/daybook.component.ts]:
```suggestion
                let branchUniqueName = this.generalService.currentBranchUniqueName ? this.generalService.currentBranchUniqueName : this.currentBranch ? this.currentBranch?.uniqueName : "";
```

File: apps/web-giddh/src/app/services/daybook.service.ts [hunk 3]
```diff
@@ -86,4 +86,27 @@ export class DaybookService {
         }),
             catchError((e) => this.errorHandler.HandleCatch<DayBookResponseModel, DayBookRequestModel>(e, request)));
     }
+
+    public ExportDaybookExpandedPost(request: any,branchUniqueName:string): Observable<BaseResponse<any, any>> {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/daybook.service.ts]: Write documentation and format the code

File: apps/web-giddh/src/app/services/apiurls/daybook.api.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 export const DAYBOOK_SEARCH_API = {
     SEARCH: 'company/:companyUniqueName/daybook?page=:page&count=:count&from=:from&to=:to', // get call
     EXPORT: 'company/:companyUniqueName/export-daybook-v2?page=:page&count=:count&from=:from&to=:to&format=:format&type=:type&sort=:sort', // get call
+    ENTRIES_EXPORT: 'company/:companyUniqueName/exports?output=base64', // get call
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/apiurls/daybook.api.ts]:
```suggestion
    ENTRIES_EXPORT: 'company/:companyUniqueName/exports?output=:output', // get call
```

File: apps/web-giddh/src/app/services/daybook.service.ts [hunk 5]
```diff
@@ -86,4 +86,27 @@ export class DaybookService {
...
+        this.companyUniqueName = this.generalService.companyUniqueName;
+        let url = this.config.apiUrl + DAYBOOK_SEARCH_API.ENTRIES_EXPORT
+            ?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))
+            ?.replace(':from', encodeURIComponent(request.from))
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/daybook.service.ts]:
```suggestion
       
```

File: apps/web-giddh/src/app/services/daybook.service.ts [hunk 6]
```diff
@@ -86,4 +86,27 @@ export class DaybookService {
...
+            ?.replace(':to', encodeURIComponent(request.to))
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/daybook.service.ts]:
```suggestion
      
```

File: apps/web-giddh/src/app/services/daybook.service.ts [hunk 7]
```diff
@@ -86,4 +86,27 @@ export class DaybookService {
...
+            ?.replace(':type', request.type?.toString())
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/daybook.service.ts]:
```suggestion
         
```

File: apps/web-giddh/src/app/services/daybook.service.ts [hunk 8]
```diff
@@ -86,4 +86,27 @@ export class DaybookService {
...
+            ?.replace(':output', request.fileType?.toString())
+        if (branchUniqueName) {
+            branchUniqueName = branchUniqueName !== this.companyUniqueName ? branchUniqueName : '';
+            url = url.concat(`&branchUniqueName=${branchUniqueName}`);
+        }
+        let format = request.fileType
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/daybook.service.ts]:
```suggestion
        const request = request.fileType;
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/daybook/daybook.component.ts (hunk 1 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/daybook/daybook.component.ts]: This code is not written by me it was written by someone five years ago.

File: apps/web-giddh/src/app/daybook/daybook.component.ts (hunk 2 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/daybook/daybook.component.ts]: done

File: apps/web-giddh/src/app/services/apiurls/daybook.api.ts (hunk 4 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/apiurls/daybook.api.ts]: done

File: apps/web-giddh/src/app/services/daybook.service.ts (hunk 6 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/daybook.service.ts]: done

File: apps/web-giddh/src/app/services/daybook.service.ts (hunk 8 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/daybook.service.ts]: The code was not used, so remove it.

File: apps/web-giddh/src/app/services/daybook.service.ts (hunk 5 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/daybook.service.ts]: done

File: apps/web-giddh/src/app/services/daybook.service.ts (hunk 7 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/daybook.service.ts]: done

File: apps/web-giddh/src/app/services/daybook.service.ts (hunk 3 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/daybook.service.ts]: done

**Review by ravinderwalkover — APPROVED** [review]

**Review by dvCodeWorld — APPROVED** [review]

**Pull request merged**